pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -41,6 +46,47 @@ func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]
 // is true, typically after load balancers have been provisioned. It can be used
 // to create DNS records.
 func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput) error {
+	gcpCluster := &capg.GCPCluster{}
+	key := client.ObjectKey{
+		Name:      in.InfraID,
+		Namespace: capiutils.Namespace,
+	}
+	if err := in.Client.Get(ctx, key, gcpCluster); err != nil {
+		return fmt.Errorf("failed to get GCP cluster: %w", err)
+	}
+
+	// public load balancer is created by CAPG. The health check for this load balancer is also created by
+	// the CAPG.
+	apiIPAddress := gcpCluster.Spec.ControlPlaneEndpoint.Host
```

Comment:
Could this be an empty string at this point? Maybe it's worth adding a debug log just in case.